src/uri.rs

```diff
@@ -419,6 +382,77 @@ impl UUri {
         }
     }
 
+    // [impl->dsn~uri-authority-name-length~1]
+    // [impl->dsn~uri-host-only~2]
+    fn verify_authority(authority: &str) -> Result<String, UUriError> {
+        Authority::try_from(authority)
+            .map_err(|e| UUriError::validation_error(format!("invalid authority: {}", e)))
+            .and_then(|auth| Self::verify_parsed_authority(&auth))
+    }
+
+    // [impl->dsn~uri-authority-name-length~1]
+    // [impl->dsn~uri-host-only~2]
+    fn verify_parsed_authority(auth: &Authority) -> Result<String, UUriError> {
+        if auth.has_port() {
+            Err(UUriError::validation_error(
+                "uProtocol URI's authority must not contain port",
+            ))
+        } else if auth.has_username() || auth.has_password() {
+            Err(UUriError::validation_error(
+                "uProtocol URI's authority must not contain userinfo",
+            ))
+        } else {
+            let auth_name = auth.host().to_string();
+            if auth_name.len() <= 128 {
+                Ok(auth_name)
+            } else {
+                Err(UUriError::validation_error(
+                    "URI's authority name must not exceed 128 characters",
+                ))
+            }
+        }
+    }
+
+    fn verify_major_version(major_version: u32) -> Result<u8, UUriError> {
+        u8::try_from(major_version).map_err(|_e| {
+            UUriError::ValidationError(
+                "uProtocol URI's major version must be an 8 bit unsigned integer".to_string(),
+            )
+        })
+    }
+
+    /// Verifies that this UUri is indeed a valid uProtocol URI.
+    ///
+    /// This check is not necessary, if any of UUri's constructors functions has been used
+    /// to create the URI. However, if the origin of a UUri is unknown, e.g. when it has
+    /// been deserialized from a protobuf, then this function can be used to check if all
+    /// properties are compliant with the uProtocol specification.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if this UUri is not a valid uProtocol URI. The returned error may
+    /// contain details regarding the cause of the validation to have failed.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use up_rust::UUri;
+    ///
+    /// let uuri = UUri {
+    ///   authority_name: "valid_name".into(),
+    ///   ue_id: 0x1000,
+    ///   ue_version_major: 0x01,
+    ///   resource_id: 0x8100,
+    ///   ..Default::default()
+    /// };
+    /// assert!(uuri.check_validity().is_ok());
+    /// ```
+    pub fn check_validity(&self) -> Result<(), UUriError> {
+        Self::verify_authority(self.authority_name.as_str())?;
+        Self::verify_major_version(self.ue_version_major)?;
```

Comment:
Could you also add a `verify_resource_id()` that'd attempt the conversion to a `u16` and include calling that function in here too?
![image](https://github.com/user-attachments/assets/1707345b-c94a-458b-a4cc-3efdd77998fd)

Reply:
Oh, absolutely. Don't know how that slipped through. Good that we do the reviews ... ;-)